clis/teliod/src/daemon.rs

```diff
@@ -47,32 +65,112 @@ fn telio_task(
+                    let new_ip_address = map
+                        .ip_addresses
+                        .as_deref()
+                        .and_then(|addresses| addresses.first().cloned());
+                    let address_changed = state.interface_ip_address != new_ip_address;
+                    state.interface_ip_address = new_ip_address;
+
+                    if state.meshnet_is_enabled {
+                        match telio.set_config(&Some(map)) {
+                            Ok(_) => {
+                                // configure the interface ip address
+                                if address_changed && auto_configure_interface {
+                                    task_set_ip(tx_channel.clone());
+                                }
+                            }
+                            Err(e) => {
+                                error!("Unable to set meshmap due to {e}");
+                            }
+                        }
                     }
                 }
                 TelioTaskCmd::GetStatus(response_tx_channel) => {
                     let status_report = TelioStatusReport {
-                        is_running: telio.is_running(),
-                        nodes: telio.external_nodes()?,
+                        telio_is_running: telio.is_running(),
+                        meshnet_enabled: state.meshnet_is_enabled,
+                        meshnet_ip: state.interface_ip_address,
+                        external_nodes: telio.external_nodes()?,
                     };
                     debug!("Telio status: {:#?}", status_report);
                     if response_tx_channel.send(status_report).is_err() {
                         error!("Telio task failed sending status report")
                     }
                 }
+                TelioTaskCmd::SetMeshnet(enabled, response_tx_channel) => {
+                    let mut result = Ok(());
+                    state.meshnet_is_enabled = enabled;
+                    if enabled {
+                        debug!("Enabling meshnet");
+
+                        // the task will call UpdateMeshmap to set_config
+                        task_retrieve_meshmap(
+                            node_identity.clone(),
+                            auth_token.clone(),
+                            tx_channel.clone(),
+                        );
+                    } else {
+                        debug!("Disabling meshnet");
+                        if let Err(e) = telio.set_config(&None) {
+                            error!("Unable to disable meshnet due to {e}");
+                            result = Err(TeliodError::DeviceError(e));
+                        }
+                    }
+                    if response_tx_channel.send(result).is_err() {
+                        error!("Telio task set meshnet failed sending result")
+                    }
+                }
+                TelioTaskCmd::SetIp(response_tx_channel) => {
```

Comment:
Isn't this a bit trivial? If we start meshnet couldn't the interface be automatically configured? the same for the opposite

Reply:
It actually does, `UpdateMeshmap` has this call
```rust
if address_changed && auto_configure_interface {
  task_set_ip(tx_channel.clone());
}
```